query-engine/connectors/sql-query-connector/src/ordering.rs

```diff
@@ -54,33 +57,36 @@ pub fn compute_joins(
     order_by: &OrderBy,
     order_by_index: usize,
     base_model: &ModelRef,
-) -> (Vec<AliasedJoin>, Column<'static>) {
+) -> (Vec<OrderingJoin>, Column<'static>) {
     let join_prefix = format!("{}{}", ORDER_JOIN_PREFIX, order_by_index);
     let mut joins = vec![];
 
-    for rf in order_by.path.iter() {
-        let (join_alias, join_data) = compute_join(base_model, rf, join_prefix.as_str());
+    if order_by.sort_aggregation.is_some() {
+        let ordering_join = compute_aggr_join(order_by, join_prefix.as_str());
 
-        joins.push(AliasedJoin {
-            data: join_data,
-            alias: join_alias,
-        });
+        joins.push(ordering_join);
```

Comment:
If there's some `sort_aggregation`, we compute a special join, and there'll only ever be one because we don't support multiple hops